unity/Assets/Scripts/ActionDispatcher.cs

```diff
@@ -322,7 +322,7 @@ public static class ActionDispatcher {
             var paramDict = methodParams.ToDictionary(param => param.Name, param => param);
             var invalidArgs = dynamicServerAction
                 .Keys()
-                .Where(argName => !paramDict.ContainsKey(argName) && argName != "action")
+                .Where(argName => !paramDict.ContainsKey(argName) && !DynamicServerAction.AllowedExtraneousParameters.Contains(argName))
```

Comment:
I like that everything is under `DynamicServerAction.AllowedExtraneousParameters` the only thing is that I feel dispatch should be called with the method `arguments` only, so by this point we shouldn't be filtering. If we did this we also need to change line `342` which passes the arguments for the exception to print the message now the message will look: `you called with arguments: 'x', 'y', 'z', 'action', 'sequenceId'`

Reply:
The problem is that we have an argument `renderInstanceSegmentation` that can be used as a global and action argument.  So we have to defer filtering.

Reply:
Your point about the exception message is good - i'll refactor to take that into account.